source/operations/external-iam/configure-keycloak-identity-management.rst

```diff
@@ -0,0 +1,259 @@
+   Defer to the :ref:`minio-installation` for more complete documentation on new MinIO deployments.
+
+   This procedure *may* work as expected for older versions of MinIO.
+
+.. cond:: not container
+
+   Keycloak Deployment and Realm Configuration
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+   This procedure assumes an existing Keycloak deployment to which you have administrative access.
+   Specifically, you must have permission to create and configure Realms, Clients, Client Scopes, Realm Roles, Users, and Groups on the Keycloak deployment.
+
+   .. cond:: k8s
+
+      For Keycloak deployments within the same Kubernetes cluster as the MinIO Tenant, this procedure assumes bidirectional access between the Keycloak and MinIO pods/services.
+
+      For Keycloak deployments external to the Kubernetes cluster, this procedure assumes an existing Ingress, Load Balancer, or similar Kubernetes network control component that manages network access to and from the MinIO Tenant.
+
+   .. cond:: not k8s
+
+      This procedure assumes bidirectional access between the Keycloak and MinIO deployments.
+
+Install and Configure ``mc`` with Access to the MinIO Cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This procedure uses :mc:`mc` for performing operations on the MinIO cluster. 
+Install ``mc`` on a machine with network access to the cluster.
+
+.. cond:: k8s
+
+   Your local host must have access to the MinIO Tenant, such as through Ingress, a Load Balancer, or a similar Kubernetes network control component.
+
+See the ``mc`` :ref:`Installation Quickstart <mc-install>` for instructions on downloading and installing ``mc``.
+
+This procedure assumes a configured :mc:`alias <mc alias>` for the MinIO cluster. 
+
+.. _minio-external-identity-management-keycloak-configure:
+
+Configure MinIO for Keycloak Identity Management
+------------------------------------------------
+
+.. cond:: linux or macos or windows
+
+   .. include:: /includes/linux/steps-configure-keycloak-identity-management.rst
+
+.. cond:: k8s
+
+   .. include:: /includes/k8s/steps-configure-keycloak-identity-management.rst
+
+.. cond:: container
+
+   .. include:: /includes/container/steps-configure-keycloak-identity-management.rst
+
+Enable the Keycloak Admin REST API
+----------------------------------
+
+MinIO supports using the Keycloak Admin REST API for checking if an authenticated user exists *and* is enabled on the Keycloak realm.
+This functionality allows MinIO to more quickly remove access from previously authenticated Keycloak users.
+
+This procedure assumes an existing MinIO deployment configured with Keycloak as an external identity manager.
+
+1) Create the Necessary Client Scopes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Navigate to the :guilabel:`Client scopes` view and create a new scope:
+
+- Set the :guilabel:`Name` to a recognizable name for the scope (``minio-admin-API-access``)
```

Comment:
This section renders as a mess of gray boxes.
Maybe field lists will improve it? Or a table?

![image](https://user-images.githubusercontent.com/53910321/222577816-3e945000-623d-4329-a4fa-67142722b2b0.png)